libraries/group_execute_provider.rb

```diff
@@ -0,0 +1,28 @@
+class Chef
+  class Provider
+    class GroupExecute < Chef::Provider::Execute
+      provides :group_execute
+
+      def_delegators :secondary_groups
+
+      private
+
+      def command
+        command = new_resource.command
+
+        if new_resource.secondary_groups
+          groups = new_resource.secondary_groups
+          if group
+            # Always do the 'primary' group last in the call chain
+            groups = [group] + groups
+          end
+          groups.each do |secondary_group|
+            command = "sg #{secondary_group} #{Shellwords.shellescape(command)}"
```

Comment:
Does this work as expected if the original command is an Array rather than a String?

Reply:
Oh, I remember seeing that was a possible type for the argument and then blocking it from my mind. No, it's not handled. Would it be OK if I just asserted it as unsupported? I don't really want to deal with escaping spaces in arguments prior to joining the array into a flat string.

Reply:
Yeah I think as long as it blows up rather than silently does something unexpected that's fine.

In vanilla Ruby I'd just raise an ArgumentError but I'm not sure if there is a more chefly way to re-declare an inherited parameter. If there's nothing obvious in the docs you've already encountered I'd suggest just dumping the following at the start of the command method.
```ruby
raise ArgumentError, "Only String commands are supported" unless command.is_a? String
```

Reply:
I think I might know a way, let me give it a shot. Otherwise, I'll just raise as you suggested.

Reply:
Alright, after b73f644, you'll now be greeted with this message:
```
 Chef::Exceptions::ValidationFailed
       ----------------------------------
       Option command must be a kind of [String]!  You passed ["echo", "foo"].
```

Reply:
Great! I've never overloaded/subclassed a chef resource like this so you're blazing trails I have not.